pymc4/distributions/tensorflow/distribution.py

```diff
@@ -0,0 +1,66 @@
+import abc
+from ...coroutine_model import Model, unpack
+from ..backend import TensorflowBackend
+
+
+class Distribution(Model, TensorflowBackend):
+
+    def __init__(self,
+                 name,
+                 keep_auxiliary=False,
+                 keep_return=True,
+                 transform=None,
+                 **kwargs):
+        super().__init__(
+            self.unpack_distribution,
+            name=name,
+            keep_auxiliary=keep_auxiliary,
+            keep_return=keep_return,
+        )
+        self._init_backend()
+        pass
+
+    def unpack_distribution(self):
+        return unpack(self)
+
+    @abc.abstractmethod
+    def _init_backend(self):
+        pass
+
+    @abc.abstractmethod
+    def sample(self, shape=(), seed=None):
```

Comment:
As mentioned above `self._backend_distribution.sample(*args, **kwargs)` shoul work in 90% cases